src/components/Table.tsx

```diff
@@ -0,0 +1,281 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint-disable prettier/prettier */
+import { Link, useParams, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { useMemo } from 'react';
+import { Person } from '../types';
+import { SortType } from '../types/SortType';
+import { SearchLink } from './SearchLink';
+
+export const Table = ({ people }: { people: Person[] }) => {
+  const param = useParams();
+  const [searchParams] = useSearchParams();
+  const activeSort = searchParams.get('sort');
+  const activeOrder = searchParams.get('order');
+  const activeSex = searchParams.get('sex');
+  const activeCenturies = searchParams.getAll('centuries');
+  const activeQuery = searchParams.get('query');
+
+  const PersonLink = ({ slug, sex, name }: Record<string, string>) => {
+    return (
+      <Link
+        to={{ pathname: `/people/${slug}`, search: searchParams.toString() }}
+        className={cn({ 'has-text-danger': sex === 'f' })}
+      >
+        {name}
+      </Link>
+    );
+  };
+
+  const computedPeople = useMemo(
+    () =>
+      people.map(el => {
+        const mother = people.find(person => person.name === el.motherName);
+        const father = people.find(person => person.name === el.fatherName);
+
+        return { ...el, mother, father };
+      }),
+    [people],
+  );
+
+  const sortedPeople = useMemo(() => {
+    const tempPeople = [...computedPeople];
+
+    switch (activeSort) {
+      case SortType.NAME:
+        tempPeople.sort((prev, next) => prev.name.localeCompare(next.name));
+        break;
+
+      case SortType.BORN:
+        tempPeople.sort((prev, next) => prev.born - next.born);
+        break;
+
+      case SortType.DIED:
+        tempPeople.sort((prev, next) => prev.died - next.died);
+        break;
+
+      case SortType.SEX:
+        tempPeople.sort((prev, next) => prev.sex.localeCompare(next.sex));
+        break;
+
+      default:
+        return tempPeople;
+    }
+
+    if (activeOrder === 'desc') {
+      tempPeople.reverse();
+    }
+
+    return tempPeople;
+  }, [computedPeople, activeSort, activeOrder]);
+
+  const filteredPeople = useMemo(() => {
```

Comment:
I'm not sure that any of these useMemo will lead to render optimisation. 
This hook uses shallow comparison for dependency array and here each dependancy array contains an array as a dependency. Which means, each time this component receives people prop => these 'memoized' values will be recalculated. 
In these cases caching looks like a more expensive calculation than the actual calculation :)

Reply:
I removed useMemo from filteredPeople becouse it doesnt change much, but leaved on sorting and computed people becouse I dont want to compute them every time I type something - especialy whe people array is very big